core/app/models/spree/calculator/weighted_rate.rb

```diff
@@ -0,0 +1,19 @@
+require_dependency 'spree/calculator'
+
+module Spree
+  class Calculator::WeightedRate < Calculator
+    preference :amount, :decimal, default: 0
+    preference :currency, :string, default: -> { Spree::Config[:currency] }
+
+    def compute_line_item(line_item)
+      if line_item && preferred_currency.casecmp(line_item.currency).zero?
```

Comment:
Yah my concern was that we're only checking _one_ of the line item currencies. When you see `line_item.currency` that doesn't guarantee that all the line items are the same currency right?  (They _are_ but the interface doesn't make that clear.)

Personally i think it'd be more intuitive to see us check `line_item.order.currency` (or `line_item.order_currency`).

But like I said, probably not worth worrying about too much.